### PR TITLE
 Use fmf to store the newly created nitrate case id 

### PR DESCRIPTION
**Pull request by jscotka**
It replaces plain text editing of fmf files for nitrate extra args by direct using of FMF
 `modify`.

It works as charm

depends on: https://github.com/psss/fmf/pull/102
solves: https://github.com/psss/tmt/issues/427

## input file
```
components: [cockpit, cockpit-doc, cockpit-system, cockpit-bridge, cockpit-ws]
version_from_comps: [cockpit-bridge]

tags+: [Tier1]
relevancy: |
   distro != rhel-7: False
   distro = rhel-7 && arch=ppc64: False
beaker-task: /CoreOS/cockpit/Sanity/selenium-generic
test: selenium-base.py:BasicTestSuite
plans: [19006]
dependencies: []

/login:
   summary: "Check cockpit basic if is able to"
   /add_ssh_key:
      relevancy: |
         distro <= rhel-7: False
         distro = rhel-7 && arch=ppc64: False
      test+:  .test15BaseSSHKeyAdded
      summary+: "add ssh key for login user"
   /login:
      /sanity:
         test+: .test20Login
         summary+: "login with various scenarios: nonexisting user, bad password, proper user"
      /special_cases:
         test: selenium-login-special.py
         summary: "Check regressions in login cases, (text in bashrc, special login shell)"
         relevancy+: |
            distro <= rhel-8: False

```

## modified file
```
components:
- cockpit
- cockpit-doc
- cockpit-system
- cockpit-bridge
- cockpit-ws
version_from_comps:
- cockpit-bridge
tags+:
- Tier1
relevancy: |
  distro != rhel-7: False
  distro = rhel-7 && arch=ppc64: False
beaker-task: /CoreOS/cockpit/Sanity/selenium-generic
test: selenium-base.py:BasicTestSuite
plans:
- 19006
dependencies: []
/login:
  summary: Check cockpit basic if is able to
  /add_ssh_key:
    relevancy: |
      distro <= rhel-7: False
      distro = rhel-7 && arch=ppc64: False
    test+: .test15BaseSSHKeyAdded
    summary+: add ssh key for login user
    extra-nitrate: TC#0609450
  /login:
    /sanity:
      test+: .test20Login
      summary+: 'login with various scenarios: nonexisting user, bad password, proper
        user'
      extra-nitrate: TC#0609451
    /special_cases:
      test: selenium-login-special.py
      summary: Check regressions in login cases, (text in bashrc, special login shell)
      relevancy+: |
        distro <= rhel-8: False
      extra-nitrate: TC#0609452
```

### CODE REVIEW

**Comment by jscotka** [comment]
@psss @lukaszachy please consider adding it to milestone 1.4

**Comment by psss** [comment]
Makes sense, let's include this in 1.4. I would just wait until the psss/fmf/pull/102 is merged and released. I think we could / should do the new `fmf` release quite soon.

File: tmt/export.py
Comment (psss) [inline tmt/export.py:234]: I guess we should modify the `dict`, right?

```suggestion
                    data['extra-nitrate'] = nitrate_case.identifier
```

Comment (jscotka) [reply tmt/export.py:234]: backward compatibility not issue.

**Comment by psss** [comment]
Improved a bit to show the info message for both implementations. @jscotka, please have a look.

**Comment by jscotka** [comment]
> Improved a bit to show the info message for both implementations. @jscotka, please have a look.

Thanks a a lot @psss , seem well

**Comment by psss** [comment]
Good, going to squash and rebase. Thanks.

**Review by psss — APPROVED** [review]

**Pull request merged**
